.travis.yml

```diff
@@ -53,10 +53,8 @@ before_script:
   - eval "${MATRIX_EVAL}"
 
 script:
-  - mkdir build
-  - cd build
-  - cmake -D ENABLE_COVERAGE:BOOL=ON ../
-  - cmake --build . -- -j2
+  - cmake -S . -B ./build -D ENABLE_COVERAGE:BOOL=ON ../
+  - cmake --build ./build -- -j2
   - ctest -j2
```

Comment:
You can use Build and Test mode to build and test the whole project in a single command:
```bash
ctest --build-and-test . ./some_new_build_directory \
      --build-generator "Unix Makefiles" -j2 -D ENABLE_COVERAGE:BOOL=ON \
      --test-command ctest -j2
```
This makes all the other cmake commands unnecessary.

Reply:
You should avoid specifying the generator if you can. This command assumes the dependency on Unix MakeFiles, so it will fail on non-unix operating systems

Reply:
Per [the docs](https://cmake.org/cmake/help/v3.11/manual/ctest.1.html#build-and-test-mode):

> The `--build-generator` option must be provided to use `--build-and-test`.

You are right that in the context of a README, it's clearer to use some variant of `cd build/ && ctest`. Outside of build-and-test mode, ctest does not provide any means of setting the build directory. Build-and-test mode is meant to give you a single command that configures, cleans, builds, and tests; it does not belong here.